### PR TITLE
Add support for Couchbase  vector store 

### PR DESCRIPTION
**Pull request by prajwal-pai77**
This PR intends to add support for couchbase as a vector store for langflow.

### CODE REVIEW

File: docs/docs/components/vector-stores.mdx [hunk 1]
```diff
@@ -26,7 +26,8 @@ The `Astra DB` initializes a vector store using Astra DB from records. It create
 - **Collection Indexing Policy:** Indexing policy for the collection.
 
 <Admonition type="note" title="Note">
-    Ensure you configure the necessary Astra DB token and API endpoint before starting.
+  Ensure you configure the necessary Astra DB token and API endpoint before
+  starting.
```
Comment (thelokeshgoel00) [inline docs/docs/components/vector-stores.mdx:30]: Stray new line

File: src/backend/base/langflow/components/vectorsearch/CouchbaseSearch.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        bucket_name: str = "",
+        scope_name: str = "",
+        collection_name: str = "",
+        index_name: str = "",
+        couchbase_connection_string: str = "",
+        couchbase_username: str = "",
+        couchbase_password: str = "",
+        search_kwargs: Optional[NestedDict] = None,
+    ) -> List[Record]:
+        search_kwargs = search_kwargs or {}
```
Comment (thelokeshgoel00) [inline src/backend/base/langflow/components/vectorsearch/CouchbaseSearch.py:60]: Search kwargs not used

**Review by thelokeshgoel00 — COMMENTED** [review]
Minor Comments, Please resolve them

File: pyproject.toml [hunk 3]
```diff
@@ -86,6 +86,7 @@ langchain-google-vertexai = "^1.0.3"
 langchain-groq = "^0.1.3"
 langchain-pinecone = "^0.1.0"
 langchain-mistralai = "^0.1.6"
+couchbase = "^4.2.1"
```
[the remark this answers is not in the document]
Comment (prajwal-pai77) [reply pyproject.toml:89]: Nah, i think for python langchain it is 4.2.1?

File: docs/docs/components/vector-stores.mdx (hunk 1 above)
Comment (prajwal-pai77) [reply docs/docs/components/vector-stores.mdx:30]: It's getting formatted due to pre commit hook which will run make format every time i commit.

File: src/backend/base/langflow/components/vectorsearch/CouchbaseSearch.py (hunk 2 above)
Comment (prajwal-pai77) [reply src/backend/base/langflow/components/vectorsearch/CouchbaseSearch.py]: Removed search_kwargs

File: docs/docs/components/vector-stores.mdx [hunk 4]
```diff
@@ -96,6 +97,45 @@ For detailed documentation and integration guides, please refer to the [Chroma C
...
+- **Input:** Search query.
+- **Embedding:** Model used in the Vector Store.
+- **Couchbase Cluster Connection String:** Cluster Connection string.
+- **Couchbase Cluster Username:** Cluster Username.
+- **Couchbase Cluster Password:** Cluster Password.
+- **Bucket Name:** Bucket identifier.
+- **Scope Name:** Scope identifier.
+- **Collection Name:** Collection identifier in Couchbase.
+- **Index Name:** Index identifier.
+- **Search Kwargs:** Additional search parameters.
```
Comment (nithishr) [inline docs/docs/components/vector-stores.mdx:135]: Please remove this if it is not getting used.

File: src/backend/base/langflow/components/vectorsearch/CouchbaseSearch.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            "couchbase_username": {"display_name": "Couchbase username","required": True},
+            "couchbase_password": {
+                "display_name": "Couchbase password",
+                "password": True,
+                "required": True
+            },
+            "bucket_name": {"display_name": "Bucket Name","required": True},
+            "scope_name": {"display_name": "Scope Name","required": True},
+            "collection_name": {"display_name": "Collection Name","required": True},
+            "index_name": {"display_name": "Index Name"},
```
Comment (nithishr) [inline src/backend/base/langflow/components/vectorsearch/CouchbaseSearch.py:38]: `index_name` should also be required.

File: src/backend/base/langflow/components/vectorsearch/__init__.py [hunk 6]
```diff
@@ -9,10 +9,12 @@
 from .VectaraSearch import VectaraSearchComponent
 from .WeaviateSearch import WeaviateSearchVectorStore
 from .pgvectorSearch import PGVectorSearchComponent
+from .Couchbase import CouchbaseSearchComponent # type: ignore
```
Comment (nithishr) [inline src/backend/base/langflow/components/vectorsearch/__init__.py:12]: Is it possible to handle this better? Ideally, you should not have to ignore this as the other imports don't

File: src/frontend/src/icons/Couchbase/couchbase.svg
Comment (nithishr) [inline src/frontend/src/icons/Couchbase/couchbase.svg:1]: Is this getting used or only `Couchbase.jsx`?

File: src/backend/base/langflow/components/vectorsearch/__init__.py (hunk 6 above)
Comment (prajwal-pai77) [reply src/backend/base/langflow/components/vectorsearch/__init__.py:12]: I get this error
src/backend/base/langflow/components/vectorsearch/__init__.py:12: error: Skipping analyzing "langflow.components.vectorsearch.Couchbase": module is installed, but missing library stubs or py.typed marker  [import-untyped]

If i do not ignore the type

File: src/backend/base/langflow/components/vectorsearch/CouchbaseSearch.py (hunk 5 above)
Comment (prajwal-pai77) [reply src/backend/base/langflow/components/vectorsearch/CouchbaseSearch.py]: Updated. thanks

File: docs/docs/components/vector-stores.mdx (hunk 4 above)
Comment (prajwal-pai77) [reply docs/docs/components/vector-stores.mdx]: Updated.thanks

File: src/backend/base/langflow/components/vectorsearch/__init__.py (hunk 6 above)
Comment (prajwal-pai77) [reply src/backend/base/langflow/components/vectorsearch/__init__.py:12]: Even other vectore stores have added this ignore in their respective files

File: src/frontend/src/icons/Couchbase/couchbase.svg
Comment (prajwal-pai77) [reply src/frontend/src/icons/Couchbase/couchbase.svg:1]: I have followed the format, similar to other vector stores which also contain the 3 files

**Pull request merged**